### DISCUSSION TITLE
First impressions

### DISCUSSION THREAD

**Discussion by mgutz** [leftwm/leftwm#773] [General]
First impressions from a long-time i3 user. I spent a half a day configuring
LeftWM (most of that time was configuring polybar).

Pros

1. Finally good looking rounded borders with picom. Could never get this work
   work with i3. I basically had to set borders to 0 in i3 and picom would shape
   the window, but then that means using dimming to differentiate
   the active window. Rounded borders are beautiful in LeftWM.

2. Intuitive layouts. `RightWiderLeftStack` and `LeftWiderRightStack` is how
   I usually work on my dual monitor setup. That is a manual process in i3
   and it gets really annoying if I move or close the main window. This is the
   main reason I'm looking for an alternative to i3.

3. Install instructions were easy to follow, maybe because I already had rust
   tools installed.

4. Gaps by default. Yay! Had to install i3-gaps fork for i3.

Cons

1. Visual window focus isn't as intuitive. In i3, I can move left, right, up,
   and down and it will focus the window in that direction. There is only
   up/down focus in LeftWM. I get why and it might just be one of those things
   I forget as I use LeftWM more.

2. Nomenclature. Was confused by tags (virtual desktop), workspaces (regions). An
   i3 workspace is a tag.

3. Default config did not have bindings for `FocusNextTag` and `FocusPreviousTag`,
   which I use often on laptops. Was hard to find in docs because I was confusing
   tags and workspaces.

4. Not able to move windows to other tags ~~and workspaces~~. BIG CON. I see a PR
   for parts of this functionality, hope it gets resolved soon.

5. Themes. 3 out of the 4 themes I tried via `leftwm-theme` would install and then
   fail on apply (version mismatch). Themes rely on their own utilities. Luckily
   I'm decent at shell scripting to read the scripts, install the utilities, and
   otherwise make them work. Will likely not use `leftwm-theme` again. Was a good 
   idea.

LeftWM is close to what I'm looking for. Good job everyone.

**Comment by hertg** [comment]
Thanks for your feedback. I have a few notes and questions about some of the cons you mention.

1. That is a frequent feedback we get from people coming from manual tilers like i3. We might introduce a feature for directional window navigation, but as far as I can tell, this is rather low priority (#171, #771). LeftWM is in its behavior more comparable to WMs like dwm (pre-defined layouts / list-based structure) than it is to i3. Due to  this fact, I think LeftWM isn't (and maybe never will be) a drop-in replacement for i3.
2. I understand that, but changing this now would be even more confusing to everyone :laughing:. Being used to it, I think the naming of tags/workspaces makes sense, once you know what they stand for.
4. What do you mean by that? We have `MoveToTag` and `MoveWindowToNextWorkspace` / `MoveWindowToPreviousWorkspace`.
5. Do you mean community themes? I kinda feel that. When I started I went straight to creating my own theme from scratch rather than using an existing one, because I didn't understand how they work. And not understanding how something works makes it tedious to fix when something goes wrong. That's why I still have hardly any knowledge of the `leftwm-theme` part of LeftWM.
